### PR TITLE
Reviews - front and back end

### PR DESCRIPTION
**Pull request by henrybv**
- Added reviews back end routes
- 3/4 done w/ reviews front-end
- Need more work on Review submission form > can maybe re-use parts of Dane's Create New Movie form

### CODE REVIEW

File: browser/js/movies/movie-item/movie-item.html [hunk 1]
```diff
@@ -6,6 +6,8 @@
 				<div class="list-group-item"><strong>{{movie.title}}</strong></div>
 				<div class="list-group-item">Category</div>
 				<div class="list-group-item">Rating: <span class="glyphicon glyphicon-star"></span><span class="glyphicon glyphicon-star"></span><span class="glyphicon glyphicon-star"></span></div>
+				<div ui-sref="addReview({reviewId: review._id})">Read reviews</div>
```
Comment (henrybv) [inline browser/js/movies/movie-item/movie-item.html]: should be state "Reviews"

File: browser/js/common/factories/reviews-factory.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+app.factory('ReviewsFactory', function($http) {
+	return {
+		fetchAllByUser: () => {
+			return $http.get('/api/users/:userId/reviews').then(reviews => {
```
Comment (jisaf) [inline browser/js/common/factories/reviews-factory.js]: you need to update the route to this: "/api/users/"+userId+"/reviews"
and where is userId coming from?

File: browser/js/common/factories/reviews-factory.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+				return reviews.data;
+			})
+		},
+		fetchAllByMovie: () => {
+			return $http.get('/api/movies/:movieId/reviews').then(reviews => {
```
Comment (jisaf) [inline browser/js/common/factories/reviews-factory.js]: same comment as above

File: browser/js/movies/movie-item/movie-item.html [hunk 4]
```diff
@@ -6,6 +6,8 @@
 				<div class="list-group-item"><strong>{{movie.title}}</strong></div>
 				<div class="list-group-item">Category: <span ng-repeat="category in movie.category">{{category.name}} <span></div>
 				<div class="list-group-item">Rating: <span class="glyphicon glyphicon-star"></span><span class="glyphicon glyphicon-star"></span><span class="glyphicon glyphicon-star"></span></div>
+				<div ui-sref="addReview({reviewId: review._id})">Read reviews</div>
+				<div>Watch trailer</div>
```
Comment (jisaf) [inline browser/js/movies/movie-item/movie-item.html]: where does this ui-sref go? addReview doesn't take a review._id.

Also, Read Review probably shouldn't point to a state called addReview.

File: browser/js/movies/movie-item/movie-item.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
-app.directive('hfMovieItem', function(MovieFactory) {
+app.directive('HfMovieItem', function(MovieFactory) {
```
Comment (jisaf) [inline browser/js/movies/movie-item/movie-item.js]: why did you change the directive name?

File: browser/js/movies/movie-item/movie-item.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
...
 	return {
 		restrict: 'E',
-		templateUrl: '/js/movies/movie-item/movie-item.html',
+		templateUrl: '/js/movies/movie-item.html',
```
Comment (jisaf) [inline browser/js/movies/movie-item/movie-item.js]: This template is in /movies/movie-item/movie-item.html

File: browser/js/reviews/review-form.html [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      <legend>New Review</legend>
+      <div ng-messages="reviewForm.title.$error" ng-hide="reviewForm.title.$pristine" role="alert">
+        <div ng-message="required" class="alert alert-warning">Review cannot be empty.</div>
+        <div ng-message="maxlength" class="alert alert-warning">Review title is too long</div>
+      </div>
+      
+      <div class="form-group">
+        <label for="title" class="col-xs-2 control-label">Title</label>
+        <div class="col-xs-10">
+          <input class="form-control" name="title" type="text" ng-model="newReviewTitle.name" required ng-maxlength="16">
```
Comment (jisaf) [inline browser/js/reviews/review-form.html]: 16 characters is pretty short. can we increase it to 80?

File: browser/js/reviews/review-form.html [hunk 8]
```diff
@@ -0,0 +1,57 @@
+<div class="well">
+  <form class="form-horizontal" novalidate name="reviewForm" ng-submit="submitReview()">
```
Comment (jisaf) [inline browser/js/reviews/review-form.html]: submitReview() isn't a function anywhere

File: browser/js/reviews/review-item/review-item.html [hunk 9]
```diff
@@ -0,0 +1,14 @@
+<!-- <div class="col-xs-6"> -->
+		<div class="well text-center" style="height: 150px">Movie image</div>
+		<div class="panel-body">
+			<div class="list-group">
+				<div class="list-group-item"><strong>{{review.title}}, {{review.movie}} by {{review.user}}</strong></div>
+				<div class="list-group-item">Created: {{review.dateCreated}}</div>
```
Comment (jisaf) [inline browser/js/reviews/review-item/review-item.html:5]: We might need to format dateCreated on the front of back end because I think momentjs will return a really long ugly format. maybe use momentjs to do relative time for added sexiness: http://momentjs.com/docs/#/displaying/fromnow/

File: browser/js/reviews/review-item/review-item.html [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+				<div class="list-group-item">Rating: {{review.rating}}<span class="glyphicon glyphicon-star"></span><span class="glyphicon glyphicon-star"></span><span class="glyphicon glyphicon-star"></span></div>
```
Comment (jisaf) [inline browser/js/reviews/review-item/review-item.html]: this shows something like Rating: 4.5**\* or are the stars are dynamically added?

File: browser/js/reviews/review-item/review-item.html [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+				<div>{{review.content}}</div>
+				<div ng-repeat="bestFor in reviews.bestFor">{{bestFor}}</div>
```
Comment (jisaf) [inline browser/js/reviews/review-item/review-item.html]: this will create lots of <divs>, might be better to ng-repeat on a <span> inside the div.

File: browser/js/reviews/review-item/review-item.js [hunk 12]
```diff
@@ -0,0 +1,9 @@
+app.directive('HfReviewItem', function(reviewFactory) {
```
Comment (jisaf) [inline browser/js/reviews/review-item/review-item.js]: probably best to stick to hf instead of Hf for consistency

Comment (jisaf) [inline browser/js/reviews/review-item/review-item.js]: also, what is reviewFactoy? did you mean ReviewsFactory? I don't know directives well so not sure what this should be.

File: browser/js/reviews/reviews-controller.js [hunk 13]
```diff
@@ -0,0 +1,7 @@
+app.controller('ReviewsCtrl', function($scope, ReviewsFactory) {
+	ReviewsFactory.fetchAll()
```
Comment (jisaf) [inline browser/js/reviews/reviews-controller.js]: fetchAll isn't a function, you only have fetchAllByMovie or ByUser

File: server/app/routes/reviews/index.js [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+const router = require('express').Router();
+const mongoose = require('mongoose');
+const Reviews = mongoose.model('Reviews');
+module.exports = router;
+
+// we are probably not using /Reviews page right now (stretch feature), unless
+// we have time to build a Recommendations page/ engine that highlights reviews/
+// movies that might interest a particular User.
+
+router.get('/', (req, res, next) => {
```
Comment (jisaf) [inline server/app/routes/reviews/index.js:13]: you don't seem to use this anywhere.

File: browser/js/reviews/review-form.html (hunk 8 above)
Comment (henrybv) [inline browser/js/reviews/review-form.html]: changed function name in ReviewsFactory from create to submitReview.

File: browser/js/reviews/review-item/review-item.html (hunk 10 above)
Comment (henrybv) [inline browser/js/reviews/review-item/review-item.html]: we need some mock reviews data for this.

File: server/app/routes/reviews/index.js (hunk 14 above)
Comment (henrybv) [inline server/app/routes/reviews/index.js:13]: new fetchAll function added in Reviews factory in addition to existing fetchAllByMovie and fetchAllByUser functions
